### PR TITLE
refactor: add workspace flag

### LINKED ISSUE #944: Update `rit rename formula` adding an optional input flag for the workspace name
## What would you like to be added

- To update the `rit rename formula` command, adding an **optional input flag** for the workspace name (when the command exists in more than one workspace), and **remove the approbation (yes/no)** when using input flags.

Reference: https://github.com/ZupIT/ritchie-cli/pull/907 (by @brunasilvazup which can provide some insights if needed).

## Why is this needed

- To avoid using prompt when using input flags.

### PR DESCRIPTION
**Pull request by lucasdittrichzup**
### Description
Add workspace flag as optional in rename formula command.

Close #944 

### How to verify it
Try renaming a formula by passing the `--workspace` flag.

### Changelog
Add workspace flag as optional in rename formula command.

---------------------------------------------

### This pull request generated the following artifacts.
To test the health and quality of this implementation, download the respective binary for your operating system, unzip and directly run the binary like the examples below.
- **Windows**
 Download the file: **[rit-windows.zip](https://github.com/ZupIT/ritchie-cli/suites/3162374088/artifacts/72771931)**
 Unzip to some folder like: `C:\home\user\downloads\pr991`
 Access the folder: `cd C:\home\user\downloads\pr991`
 Directly call the binary: `.\rit.exe --version` or `.\rit.exe name of formula`
 

 - **Linux**
 Download the file: **[rit-linux.zip](https://github.com/ZupIT/ritchie-cli/suites/3162374088/artifacts/72771929)**
 Unzip to some folder like: `/home/user/downloads/pr991`
 Access the folder: `cd /home/user/downloads/pr991`
 Assign execute permission to binary: `chmod +x ./rit`
 Directly call the binary: `./rit --version` or `./rit name of formula`
 

 - **MacOS**
 Download the file: **[rit-macos.zip](https://github.com/ZupIT/ritchie-cli/suites/3162374088/artifacts/72771930)**
 Unzip to some folder like: `/home/user/downloads/pr991`
 Access the folder: `cd /home/user/downloads/pr991`
 Assign execute permission to binary: `chmod +x ./rit`
 Directly call the binary: `./rit --version` or `./rit name of formula`

> Generated at Mon Jul 05 2021 17:13:02 GMT+0000 (Coordinated Universal Time)

---------------------------------------------

### This pull request generated the following artifacts.
To test the health and quality of this implementation, download the respective binary for your operating system, unzip and directly run the binary like the examples below.
- **Windows**
 Download the file: **[rit-windows.zip](https://github.com/ZupIT/ritchie-cli/suites/3163399498/artifacts/72802825)**
 Unzip to some folder like: `C:\home\user\downloads\pr991`
 Access the folder: `cd C:\home\user\downloads\pr991`
 Directly call the binary: `.\rit.exe --version` or `.\rit.exe name of formula`
 

 - **Linux**
 Download the file: **[rit-linux.zip](https://github.com/ZupIT/ritchie-cli/suites/3163399498/artifacts/72802823)**
 Unzip to some folder like: `/home/user/downloads/pr991`
 Access the folder: `cd /home/user/downloads/pr991`
 Assign execute permission to binary: `chmod +x ./rit`
 Directly call the binary: `./rit --version` or `./rit name of formula`
 

 - **MacOS**
 Download the file: **[rit-macos.zip](https://github.com/ZupIT/ritchie-cli/suites/3163399498/artifacts/72802824)**
 Unzip to some folder like: `/home/user/downloads/pr991`
 Access the folder: `cd /home/user/downloads/pr991`
 Assign execute permission to binary: `chmod +x ./rit`
 Directly call the binary: `./rit --version` or `./rit name of formula`

> Generated at Mon Jul 05 2021 20:20:32 GMT+0000 (Coordinated Universal Time)

### CODE REVIEW

File: pkg/cmd/rename_formula.go [hunk 1]
```diff
@@ -50,10 +50,11 @@ const (
 
 	questionConfirmation = "Are you sure you want to rename the formula from %q to %q?"
 
-	errNonExistFormula = "Formula %q wasn't found in the workspaces"
-	errFormulaExists   = "Formula %q already exists on this workspace = %q"
-
-	renameSuccessMsg = "The formula was renamed with success"
+	errNonExistFormula = "formula %q wasn't found in the workspaces"
+	errFormulaExists   = "formula %q already exists on this workspace = %q"
+	errFormulaInManyWS = "formula %q was found in %d workspace. Please enter a value for the 'workspace' flag"
```
Comment (GuillaumeFalourd) [inline pkg/cmd/rename_formula.go:55]: Add a `s` to `workspace` as this message will only shows if more than 1 workspace is detected.

File: pkg/cmd/rename_formula.go [hunk 2]
```diff
@@ -50,10 +50,11 @@ const (
...
+	errInvalidWS       = "workspace %q was not found"
+	renameSuccessMsg   = "The formula was renamed with success"
```
Comment (GuillaumeFalourd) [inline pkg/cmd/rename_formula.go:57]: It seems we commonly use `successfully` inside the repository, therefore I suggest to update to `The formula was successfully renamed`

File: pkg/cmd/rename_formula_test.go [hunk 3]
```diff
@@ -295,6 +289,46 @@ func TestRenameFormulaCmd(t *testing.T) {
...
+			name: "err when new formula exists in two workspaces and workspace flag is empty",
+			in: in{
+				customWorkspaceSelected: true,
+				args: []string{
+					"--oldName=rit testing formula",
+					"--newName=rit testing other",
+				},
+			},
+			out: out{
+				want: errors.New("formula \"rit testing formula\" was found in 2 workspace. Please enter a value for the 'workspace' flag"),
```
Comment (GuillaumeFalourd) [inline pkg/cmd/rename_formula_test.go:315]: You'll have to add the `s` to workspace here if you update the error message I commented above.

File: pkg/formula/runner/docker/pre_run.go [hunk 4]
```diff
@@ -63,7 +63,7 @@ var (
 	)
 
 	dockerVersionRegexp *regexp.Regexp
-	once sync.Once
+	once                sync.Once
```
Comment (GuillaumeFalourd) [inline pkg/formula/runner/docker/pre_run.go:66]: 🤨

File: pkg/cmd/rename_formula.go (hunk 1 above)
Comment (lucasdittrichzup) [reply pkg/cmd/rename_formula.go]: done

File: pkg/formula/runner/docker/pre_run.go (hunk 4 above)
Comment (lucasdittrichzup) [reply pkg/formula/runner/docker/pre_run.go:66]: this addition was VSCode's default lint

File: pkg/cmd/rename_formula.go [hunk 5]
```diff
@@ -50,10 +50,11 @@ const (
...
 
-	errNonExistFormula = "Formula %q wasn't found in the workspaces"
-	errFormulaExists   = "Formula %q already exists on this workspace = %q"
-
-	renameSuccessMsg = "The formula was renamed with success"
+	errNonExistFormula = "formula %q wasn't found in the workspaces"
+	errFormulaExists   = "formula %q already exists on this workspace = %q"
+	errFormulaInManyWS = "formula %q was found in %d workspaces. Please enter a value for the 'workspace' flag"
+	errInvalidWS       = "workspace %q was not found"
+	renameSuccessMsg   = "The formula was renamed with successfully"
```
Comment (GuillaumeFalourd) [inline pkg/cmd/rename_formula.go:57]: `The formula was successfully renamed` (Ritchie convention)

File: testdata/gha_workflows/formula_workflow/assert3.txt [hunk 6]
```diff
@@ -1 +1 @@
-The formula was renamed with success
+The formula was renamed with successfully
```
Comment (GuillaumeFalourd) [inline testdata/gha_workflows/formula_workflow/assert3.txt:1]: `The formula was successfully renamed` (Ritchie convention)

File: pkg/cmd/rename_formula.go (hunk 5 above)
Comment (lucasdittrichzup) [reply pkg/cmd/rename_formula.go]: done!

Comment (lucasdittrichzup) [reply pkg/cmd/rename_formula.go]: done!

File: testdata/gha_workflows/formula_workflow/assert3.txt (hunk 6 above)
Comment (lucasdittrichzup) [reply testdata/gha_workflows/formula_workflow/assert3.txt]: done!

**Review by GuillaumeFalourd — APPROVED** [review]
Good job 👍🏼

**Pull request merged**